sql: TestShowRangesMultipleStores failed

sql.TestShowRangesMultipleStores [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4247676&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4247676&tab=artifacts#/) on master @ [5b871267b62d22854f8c7b92c7c5843100597976](https://github.com/cockroachdb/cockroach/commits/5b871267b62d22854f8c7b92c7c5843100597976):


```
=== RUN   TestShowRangesMultipleStores
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/233e6aa31ff7ad29c896f0479cb5d0c4/logTestShowRangesMultipleStores703273161
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestShowRangesMultipleStores
    show_ranges_test.go:137: -- test log scope end --
--- FAIL: TestShowRangesMultipleStores (2.24s)
=== RUN   TestShowRangesMultipleStores/SHOW_RANGES_FROM_DATABASE_system
    show_ranges_test.go:127: query '
        SELECT DISTINCT
        		(
        		array_position(replica_localities, lease_holder_locality)
        		= array_position(replicas, lease_holder)
        		)
        	FROM [SHOW RANGES FROM DATABASE system]': expected:
        true
        
        got:
        true
        NULL
        
    --- FAIL: TestShowRangesMultipleStores/SHOW_RANGES_FROM_DATABASE_system (0.12s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowRangesMultipleStores.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/show_ranges_test.go
+++ b/pkg/sql/show_ranges_test.go
@@ -124,7 +124,8 @@ func TestShowRangesMultipleStores(t *testing.T) {
 		"SHOW RANGE FROM INDEX system.jobs@jobs_status_created_idx FOR ROW ('running', now(), 0)",
 	} {
 		t.Run(q, func(t *testing.T) {
-			sqlDB.CheckQueryResults(t,
+			// Retry because if there's not a leaseholder, you can NULL.
+			sqlDB.CheckQueryResultsRetry(t,
 				fmt.Sprintf(`
 SELECT DISTINCT
 		(

